src/server/scripts/EasternKingdoms/BlackrockMountain/BlackwingLair/boss_vaelastrasz.cpp

```diff
@@ -188,24 +191,19 @@ class boss_vaelastrasz : public CreatureScript
                         break;
                     case EVENT_BURNINGADRENALINE_CASTER:
                         {
-                            Unit* target = NULL;
-
-                            uint8 i = 0;
-                            while (i < 3)   // max 3 tries to get a random target with power_mana
+                            //selects a random target that isn't the current victim and is a mana user (selects mana users) but not pets
+                            //it also ignores targets who have the aura. We don't want to place the debuff on the same target twice.
+                            if (Unit *target = SelectTarget(SELECT_TARGET_RANDOM, 1, [&](Unit* u) { return !u->IsPet() && u && u->getPowerType() == POWER_MANA && !u->HasAura(SPELL_BURNINGADRENALINE); }))
```

Comment:
this code will trigger a static analysis issue about null checking "u" after it has been dereferenced